### PR TITLE
Add `HTTP Token authentication` section to guide [skip ci]

### PR DESCRIPTION
**Pull request by soartec-lab**
### Summary
I added it because there was no guide explaining how to use HTTP Token authentication.
so please check.

https://github.com/rails/rails/blob/master/actionpack/lib/action_controller/metal/http_authentication.rb#L405

https://github.com/rails/rails/blob/master/guides/source/action_controller_overview.md#http-authentications

### CODE REVIEW

File: guides/source/action_controller_overview.md
Comment (erichmachado) [inline guides/source/action_controller_overview.md]:
```suggestion
HTTP token authentication is a scheme to enable the usage of Bearer tokens in the HTTP `Authorization` header. There are many token formats available and describing them is outside the scope of this document.

As an example, suppose you want to use an authentication token that has been issued in advance to perform authentication and access. Implementing token authentication with Rails is quite easy and only requires using one method, `authenticate_or_request_with_http_token`.
```

**Review by erichmachado — COMMENTED** [review]
First of all, thank you for working on this! It would be really great to have this covered on the guides.

I've added some suggestions to the best of my knowledge. Hopefully it will be of help.

**Comment by soartec-lab** [comment]
@erichmachado 

Thank you for check and suggestions.
I accepted and reflected your suggestions. 
Also, I did a commit squash, so please merge if there is no problem.

**Review by erichmachado — APPROVED** [review]

**Comment by erichmachado** [comment]
Hi @soartec-lab, I'm glad my feedback was helpful. I don't have merge permission, but I've gave my approval for it. Hopefully this will help move the needle a little bit.

**Comment by soartec-lab** [comment]
@kamipo 
Thanks for merge.

@erichmachado 
merged. Thank you for your support.

**Pull request merged**
